### DISCUSSION TITLE
How to use JetBrain's remote ssh functionality

### DISCUSSION THREAD

**Discussion by cuichenli** [lima-vm/lima#2452] [Q&A] [answered]
I have the following settings:

![image](https://github.com/lima-vm/lima/assets/29245119/59ed8b61-ff33-4bcb-b421-399422e1beb0)
But if I clicked the `Test Connection`, it will say connection refused. 

![image](https://github.com/lima-vm/lima/assets/29245119/549c5e82-3f35-4aac-88a8-8fbe5e1f760b)

I have checked from CLI that I can directly ssh into this VM via `ssh lima-ubuntu`.

Thanks!

**Comment by balajiv113** [comment]
Hi, Are you using the SSH Remote development ??

If Yes, Please use the host as `127.0.0.1`. With this it should work fine.

DNS lima-ubuntu is internal one and host doesn't know this

**Reply by afbjorklund** [reply, marked as answer]
If "lima-ubuntu" is available in the ssh config, then it was supposed to pick that up.

Probably should just use the default port (22), the random one will be in the config.

> Please use the host as 127.0.0.1. With this it should work fine.

It still needs to provide a valid identity (key), from somewhere.

**Reply by cuichenli** [reply]
OK I figured it out. I previously dumped the ssh config generated by lima to my `~/.ssh/config` file and then I recreated the vm a few times. Now the port number in this file is out dated, and once I updated it it is working now. 

Sorry for the spam.  🙇

**Reply by afbjorklund** [reply]
That is why you want to use the ssh config (alias), so that you don't have to update the port number manually each reboot...

Also to get all the other performance enhancements, with crypto and ssh-mux.
